lib/babe/verify.go

```diff
@@ -58,12 +60,26 @@ type VerificationManager struct {
 
 // NewVerificationManager returns a new NewVerificationManager
 func NewVerificationManager(blockState BlockState, epochState EpochState) (*VerificationManager, error) {
-	if blockState == nil {
-		return nil, errNilBlockState
+	switch blockState.(type) {
+	case *state.BlockState:
+		if blockState == (*state.BlockState)(nil) {
+			return nil, errNilBlockState
+		}
+	default:
+		if reflect.ValueOf(blockState).IsNil() {
+			return nil, errNilBlockState
+		}
 	}
```

Comment:
supernit to reduce code duplication if we add more cases :wink: 

```suggestion
	var isNil bool
	switch blockState.(type) {
	case *state.BlockState:
		isNil = blockState == (*state.BlockState)(nil)
	default:
		isNil = reflect.ValueOf(blockState).IsNil()
	}
	if isNil {
		return nil, errNilBlockState
	}
```

Reply:
updated.